connect_ferc1_to_eia.py

```diff
@@ -42,497 +42,785 @@
+                'record_id': 'record_id_ferc1', })
+            .set_index('record_id_ferc1')
+            .assign(
+                fuel_cost_per_mmbtu=lambda x: (
+                    x.total_fuel_cost / x.total_mmbtu),
+                heat_rate_mmbtu_mwh=lambda x: (
+                    x.total_mmbtu / x.net_generation_mwh),
+                plant_id_report_year=lambda x: x.plant_id_pudl.map(
+                    str) + "_" + x.report_year.map(str)
+            ))
+        if 0.9 > (len(self.steam_df) /
+                  len(self.steam_df.drop_duplicates(
+                      subset=['report_year',
+                              'utility_id_pudl',
+                              'plant_id_ferc1'])) < 1.1):
+            raise AssertionError(
+                'Merge issue with pudl_outs plants_steam_ferc1 and fbp_ferc1')
+
+        return self.steam_df
+
+    def get_train_records(self, dataset_df, dataset_id_col):
+        """
+        Generate a set of known connections from a dataset using training data.
+
+        This method grabs only the records from the the datasets (EIA or FERC)
+        that we have in our training data.
+
+        This method relies on:
+        * train_df (pandas.DataFrame): training data/known matches between ferc
+          and the master unit list. Result of `prep_train_connections()`.
+
+        Args:
+            dataset_df (pandas.DataFrame): either FERC1 steam table (result of
+                `prep_ferc_data()`) or EIA master unit list (result of
+                `get_master_unit_list_eia()`).
+            dataset_id_col (string): either `record_id_eia` for
+                plant_parts_true_df or `record_id_ferc1` for steam_df.
+
+        """
+        known_df = (pd.merge(dataset_df,
+                             self.train_df.reset_index()[[dataset_id_col]],
+                             left_index=True,
+                             right_on=[dataset_id_col]
+                             )
+                    .drop_duplicates(subset=[dataset_id_col])
+                    .set_index(dataset_id_col)
+                    .astype({'total_fuel_cost': float,
+                             'total_mmbtu': float}))
+        return known_df
+
+    def make_features(self, ferc1_df, eia_df, block_col=None):
+        """
+        Generate comparison features based on defined features.
+
+        The recordlinkage package helps us create feature vectors.
+        For each column that we have in both datasets, this method generates
+        a column of feature vecotrs, which contain values between 0 and 1 that
+        are measures of the similarity between each datapoint the two datasets
+        (1 meaning the two datapoints were exactly the same and 0 meaning they
+        were not similar at all).
+
+        For more details see recordlinkage's documentaion:
+        https://recordlinkage.readthedocs.io/en/latest/ref-compare.html
+
+        TODO: Should this be a @staticmethod?
```

Comment:
Not unless you specifically need to be able to call it without instantiating the object.